File: Test/AdventureWorksFunctionalModel/AdventureWorksEFCoreContext.cs
using AW.Mapping;
using AW.Types;
using Microsoft.EntityFrameworkCore;

namespace AW {
    public class AdventureWorksEFCoreContext : DbContext {
        private readonly string _nameOrConnectionString;

        public AdventureWorksEFCoreContext(string nameOrConnectionString) => _nameOrConnectionString = nameOrConnectionString;

        public DbSet<Department> Departments { get; init; }

        //public DbSet<Employee> Employees { get; init; }
        //public DbSet<EmployeeAddress> EmployeeAddresses { get; init; }
        //public DbSet<EmployeeDepartmentHistory> EmployeeDepartmentHistories { get; init; }
        //public DbSet<EmployeePayHistory> EmployeePayHistories { get; init; }
        //public DbSet<JobCandidate> JobCandidates { get; init; }
        public DbSet<Shift> Shifts { get; init; }

        //public DbSet<Address> Addresses { get; init; }
        public DbSet<AddressType> AddressTypes { get; init; }

        //public DbSet<BusinessEntity> BusinessEntities { get; init; }
        //public DbSet<BusinessEntityAddress> BusinessEntityAddresses { get; init; }
        //public DbSet<BusinessEntityContact> BusinessEntityContacts { get; init; }
        //public DbSet<Person> Contacts { get; init; }
        public DbSet<ContactType> ContactTypes { get; init; }

        public DbSet<CountryRegion> CountryRegions { get; init; }

        //public DbSet<PersonPhone> PersonPhones { get; init; }
        public DbSet<PhoneNumberType> PhoneNumberTypes { get; init; }

        //public DbSet<EmailAddress> EmailAddresses { get; init; }
        public DbSet<Password> Passwords { get; init; }
        public DbSet<StateProvince> StateProvinces { get; init; }
        public DbSet<BillOfMaterial> BillOfMaterials { get; init; }
        public DbSet<Culture> Cultures { get; init; }
        public DbSet<Illustration> Illustrations { get; init; }
        public DbSet<Location> Locations { get; init; }
        public DbSet<Product> Products { get; init; }
        public DbSet<ProductCategory> ProductCategories { get; init; }
        public DbSet<ProductCostHistory> ProductCostHistories { get; init; }
        public DbSet<ProductDescription> ProductDescriptions { get; init; }
        public DbSet<ProductInventory> ProductInventories { get; init; }
        public DbSet<ProductListPriceHistory> ProductListPriceHistories { get; init; }
        public DbSet<ProductModel> ProductModels { get; init; }
        public DbSet<ProductModelIllustration> ProductModelIllustrations { get; init; }
        public DbSet<ProductModelProductDescriptionCulture> ProductModelProductDescriptionCultures { get; init; }
        public DbSet<ProductPhoto> ProductPhotoes { get; init; }
        public DbSet<ProductProductPhoto> ProductProductPhotoes { get; init; }
        public DbSet<ProductReview> ProductReviews { get; init; }
        public DbSet<ProductSubcategory> ProductSubcategories { get; init; }
        public DbSet<ScrapReason> ScrapReasons { get; init; }
        public DbSet<TransactionHistory> TransactionHistories { get; init; }
        public DbSet<UnitMeasure> UnitMeasures { get; init; }
        public DbSet<WorkOrder> WorkOrders { get; init; }

        public DbSet<WorkOrderRouting> WorkOrderRoutings { get; init; }

        //public DbSet<ProductVendor> ProductVendors { get; init; }
        //public DbSet<PurchaseOrderDetail> PurchaseOrderDetails { get; init; }
        //public DbSet<PurchaseOrderHeader> PurchaseOrderHeaders { get; init; }
        //public DbSet<ShipMethod> ShipMethods { get; init; }
        //public DbSet<Vendor> Vendors { get; init; }
        public DbSet<PersonCreditCard> PersonCreditCards { get; init; }

        //public DbSet<CountryRegionCurrency> CountryRegionCurrencies { get; init; }
        public DbSet<CreditCard> CreditCards { get; init; }

        public DbSet<Currency> Currencies { get; init; }

        //public DbSet<CurrencyRate> CurrencyRates { get; init; }
        //public DbSet<Customer> Customers { get; init; }
        //public DbSet<SalesOrderDetail> SalesOrderDetails { get; init; }
        //public DbSet<SalesOrderHeader> SalesOrderHeaders { get; init; }
        //public DbSet<SalesOrderHeaderSalesReason> SalesOrderHeaderSalesReasons { get; init; }
        //public DbSet<SalesPerson> SalesPersons { get; init; }
        //public DbSet<SalesPersonQuotaHistory> SalesPersonQuotaHistories { get; init; }
        //public DbSet<SalesReason> SalesReasons { get; init; }
        //public DbSet<SalesTaxRate> SalesTaxRates { get; init; }
        public DbSet<SalesTerritory> SalesTerritories { get; init; }

        //public DbSet<SalesTerritoryHistory> SalesTerritoryHistories { get; init; }
        //public DbSet<ShoppingCartItem> ShoppingCartItems { get; init; }
        public DbSet<SpecialOffer> SpecialOffers { get; init; }

        public DbSet<SpecialOfferProduct> SpecialOfferProducts { get; init; }
        //public DbSet<Store> Stores { get; init; }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder) {
            optionsBuilder.UseSqlServer(_nameOrConnectionString);
            optionsBuilder.UseLazyLoadingProxies();
            //optionsBuilder.LogTo(m => Debug.WriteLine(m), LogLevel.Trace);
        }

        protected override void OnModelCreating(ModelBuilder modelBuilder) {
            modelBuilder.Entity<Department>().Map();
            modelBuilder.Entity<Employee>().Map();
            modelBuilder.Entity<EmployeeDepartmentHistory>().Map();
            modelBuilder.Entity<EmployeePayHistory>().Map();
            modelBuilder.Entity<JobCandidate>().Map();
            modelBuilder.Entity<Shift>().Map();
            modelBuilder.Entity<Address>().Map();
            modelBuilder.Entity<AddressType>().Map();
            modelBuilder.Entity<BusinessEntity>().Map();
            modelBuilder.Entity<BusinessEntityAddress>().Map();
            modelBuilder.Entity<BusinessEntityContact>().Map();
            modelBuilder.Entity<Person>().Map();
            modelBuilder.Entity<ContactType>().Map();
            modelBuilder.Entity<CountryRegion>().Map();
            modelBuilder.Entity<EmailAddress>().Map();
            modelBuilder.Entity<Password>().Map();
            modelBuilder.Entity<PersonPhone>().Map();
            modelBuilder.Entity<PhoneNumberType>().Map();
            modelBuilder.Entity<StateProvince>().Map();
            modelBuilder.Entity<BillOfMaterial>().Map();
            modelBuilder.Entity<Culture>().Map();
            modelBuilder.Entity<Illustration>().Map();
            modelBuilder.Entity<Location>().Map();
            modelBuilder.Entity<Product>().Map();
            modelBuilder.Entity<ProductCategory>().Map();
            modelBuilder.Entity<ProductCostHistory>().Map();
            modelBuilder.Entity<ProductDescription>().Map();
            modelBuilder.Entity<ProductInventory>().Map();
            modelBuilder.Entity<ProductListPriceHistory>().Map();
            modelBuilder.Entity<ProductModel>().Map();
            modelBuilder.Entity<ProductModelIllustration>().Map();
            modelBuilder.Entity<ProductModelProductDescriptionCulture>().Map();
            modelBuilder.Entity<ProductPhoto>().Map();
            modelBuilder.Entity<ProductProductPhoto>().Map();
            modelBuilder.Entity<ProductReview>().Map();
            modelBuilder.Entity<ProductSubcategory>().Map();
            modelBuilder.Entity<ScrapReason>().Map();
            modelBuilder.Entity<TransactionHistory>().Map();
            modelBuilder.Entity<UnitMeasure>().Map();
            modelBuilder.Entity<WorkOrder>().Map();
            modelBuilder.Entity<WorkOrderRouting>().Map();
            modelBuilder.Entity<ProductVendor>().Map();
            modelBuilder.Entity<PurchaseOrderDetail>().Map();
            modelBuilder.Entity<PurchaseOrderHeader>().Map();
            modelBuilder.Entity<ShipMethod>().Map();
            modelBuilder.Entity<Vendor>().Map();
            modelBuilder.Entity<PersonCreditCard>().Map();
            modelBuilder.Entity<CountryRegionCurrency>().Map();
            modelBuilder.Entity<CreditCard>().Map();
            modelBuilder.Entity<Currency>().Map();
            modelBuilder.Entity<CurrencyRate>().Map();
            modelBuilder.Entity<Customer>().Map();
            modelBuilder.Entity<SalesOrderDetail>().Map();
            modelBuilder.Entity<SalesOrderHeader>().Map();
            modelBuilder.Entity<SalesOrderHeaderSalesReason>().Map();
            modelBuilder.Entity<SalesPerson>().Map();
            modelBuilder.Entity<SalesPersonQuotaHistory>().Map();
            modelBuilder.Entity<SalesReason>().Map();
            modelBuilder.Entity<SalesTaxRate>().Map();
            modelBuilder.Entity<SalesTerritory>().Map();
            modelBuilder.Entity<SalesTerritoryHistory>().Map();
            modelBuilder.Entity<ShoppingCartItem>().Map();
            modelBuilder.Entity<SpecialOffer>().Map();
            modelBuilder.Entity<SpecialOfferProduct>().Map();
            modelBuilder.Entity<Store>().Map();
        }
    }
}

File: Test/NOF2.Demo.Model/AdventureWorksEFCoreContext.vb
'====================================================================================================
'The Free Edition of Instant VB limits conversion output to 100 lines per file.

'To purchase the Premium Edition, visit our website:
'https://www.tangiblesoftwaresolutions.com/order/order-instant-vb.html
'====================================================================================================

Imports AW.Mapping

Imports Microsoft.EntityFrameworkCore

Namespace AW
	Public Class AdventureWorksEFCoreContext
		Inherits DbContext

		Private ReadOnly _nameOrConnectionString As String

		Public Sub New(ByVal nameOrConnectionString As String)
			_nameOrConnectionString = nameOrConnectionString
		End Sub
		Public Property Departments() As DbSet(Of Department)?
		Public Property Shifts() As DbSet(Of Shift)?
		Public Property AddressTypes() As DbSet(Of AddressType)?
		Public Property ContactTypes() As DbSet(Of ContactType)?
		Public Property CountryRegions() As DbSet(Of CountryRegion)?
		Public Property PhoneNumberTypes() As DbSet(Of PhoneNumberType)?
		Public Property Passwords() As DbSet(Of Password)?
		Public Property StateProvinces() As DbSet(Of StateProvince)?
		Public Property BillOfMaterials() As DbSet(Of BillOfMaterial)?
		Public Property Cultures() As DbSet(Of Culture)?
		Public Property Illustrations() As DbSet(Of Illustration)?
		Public Property Locations() As DbSet(Of Location)?
		Public Property Products() As DbSet(Of Product)?
		Public Property ProductCategories() As DbSet(Of ProductCategory)?
		Public Property ProductCostHistories() As DbSet(Of ProductCostHistory)?
		Public Property ProductDescriptions() As DbSet(Of ProductDescription)?
		Public Property ProductInventories() As DbSet(Of ProductInventory)?
		Public Property ProductListPriceHistories() As DbSet(Of ProductListPriceHistory)?
		Public Property ProductModels() As DbSet(Of ProductModel)?
		Public Property ProductModelIllustrations() As DbSet(Of ProductModelIllustration)?
		Public Property ProductModelProductDescriptionCultures() As DbSet(Of ProductModelProductDescriptionCulture)?
		Public Property ProductPhotoes() As DbSet(Of ProductPhoto)?
		Public Property ProductProductPhotoes() As DbSet(Of ProductProductPhoto)?
		Public Property ProductReviews() As DbSet(Of ProductReview)?
		Public Property ProductSubcategories() As DbSet(Of ProductSubcategory)?
		Public Property ScrapReasons() As DbSet(Of ScrapReason)?
		Public Property TransactionHistories() As DbSet(Of TransactionHistory)?
		Public Property UnitMeasures() As DbSet(Of UnitMeasure)?
		Public Property WorkOrders() As DbSet(Of WorkOrder)?
		Public Property WorkOrderRoutings() As DbSet(Of WorkOrderRouting)?
		Public Property PersonCreditCards() As DbSet(Of PersonCreditCard)?
		Public Property CreditCards() As DbSet(Of CreditCard)?
		Public Property Currencies() As DbSet(Of Currency)?
		Public Property SalesTerritories() As DbSet(Of SalesTerritory)?
		Public Property SpecialOffers() As DbSet(Of SpecialOffer)?
		Public Property SpecialOfferProducts() As DbSet(Of SpecialOfferProduct)?

		Protected Overrides Sub OnConfiguring(ByVal optionsBuilder As DbContextOptionsBuilder)
			optionsBuilder.UseSqlServer(_nameOrConnectionString)
			optionsBuilder.UseLazyLoadingProxies()
			'optionsBuilder.LogTo(m => Debug.WriteLine(m), LogLevel.Trace);
		End Sub

		Protected Overrides Sub OnModelCreating(ByVal modelBuilder As ModelBuilder)
			modelBuilder.Entity(Of Department)().Map()
			modelBuilder.Entity(Of Employee)().Map()
			modelBuilder.Entity(Of EmployeeDepartmentHistory)().Map()
			modelBuilder.Entity(Of EmployeePayHistory)().Map()
			modelBuilder.Entity(Of JobCandidate)().Map()
			modelBuilder.Entity(Of Shift)().Map()
			modelBuilder.Entity(Of Address)().Map()
			modelBuilder.Entity(Of AddressType)().Map()
			modelBuilder.Entity(Of BusinessEntity)().Map()
			modelBuilder.Entity(Of BusinessEntityAddress)().Map()
			modelBuilder.Entity(Of BusinessEntityContact)().Map()
			modelBuilder.Entity(Of Person)().Map()
			modelBuilder.Entity(Of ContactType)().Map()
			modelBuilder.Entity(Of CountryRegion)().Map()
			modelBuilder.Entity(Of EmailAddress)().Map()
			modelBuilder.Entity(Of Password)().Map()
			modelBuilder.Entity(Of PersonPhone)().Map()
			modelBuilder.Entity(Of PhoneNumberType)().Map()
			modelBuilder.Entity(Of StateProvince)().Map()
			modelBuilder.Entity(Of BillOfMaterial)().Map()
			modelBuilder.Entity(Of Culture)().Map()
			modelBuilder.Entity(Of Illustration)().Map()
			modelBuilder.Entity(Of Location)().Map()
			modelBuilder.Entity(Of Product)().Map()
			modelBuilder.Entity(Of ProductCategory)().Map()
			modelBuilder.Entity(Of ProductCostHistory)().Map()
			modelBuilder.Entity(Of ProductDescription)().Map()
			modelBuilder.Entity(Of ProductInventory)().Map()
			modelBuilder.Entity(Of ProductListPriceHistory)().Map()
			modelBuilder.Entity(Of ProductModel)().Map()
			modelBuilder.Entity(Of ProductModelIllustration)().Map()
			modelBuilder.Entity(Of ProductModelProductDescriptionCulture)().Map()
			modelBuilder.Entity(Of ProductPhoto)().Map()
			modelBuilder.Entity(Of ProductProductPhoto)().Map()
			modelBuilder.Entity(Of ProductReview)().Map()
			modelBuilder.Entity(Of ProductSubcategory)().Map()
			modelBuilder.Entity(Of ScrapReason)().Map()
			modelBuilder.Entity(Of TransactionHistory)().Map()
			modelBuilder.Entity(Of UnitMeasure)().Map()
			modelBuilder.Entity(Of WorkOrder)().Map()
			modelBuilder.Entity(Of WorkOrderRouting)().Map()
			modelBuilder.Entity(Of ProductVendor)().Map()
			modelBuilder.Entity(Of PurchaseOrderDetail)().Map()
			modelBuilder.Entity(Of PurchaseOrderHeader)().Map()
			modelBuilder.Entity(Of ShipMethod)().Map()
			modelBuilder.Entity(Of Vendor)().Map()
			modelBuilder.Entity(Of PersonCreditCard)().Map()
			modelBuilder.Entity(Of CountryRegionCurrency)().Map()
			modelBuilder.Entity(Of CreditCard)().Map()
			modelBuilder.Entity(Of Currency)().Map()

'====================================================================================================
'End of the allowed output for the Free Edition of Instant VB.

'To purchase the Premium Edition, visit our website:
'https://www.tangiblesoftwaresolutions.com/order/order-instant-vb.html
'====================================================================================================
